lib/Modules/ADCharmUtils/ADCharmUtils.psm1

```diff
@@ -150,43 +146,49 @@ function Invoke-DJoin {
         [Parameter(Mandatory=$true)]
         [String]$DCAddress,
         [Parameter(Mandatory=$true)]
-        [String]$DJoinBlob
+        [String]$DJoinBlob,
+        [Parameter(Mandatory=$true)]
+        [String]$DomainSuffix
     )
 
     Write-JujuWarning "Started join domain"
-
-    $networkName = (Get-MainNetadapter)
-    Set-DnsClientServerAddress -InterfaceAlias $networkName -ServerAddresses $DCAddress
+    Set-DnsClientServerAddress -InterfaceAlias * -ServerAddresses $DCAddress
+    Set-DnsClientGlobalSetting -SuffixSearchList @($DomainSuffix)
     $cmd = @("ipconfig", "/flushdns")
     Invoke-JujuCommand -Command $cmd
-
     $blobFile = Join-Path $env:TMP "djoin-blob.txt"
-    Write-FileFromBase64 -File $blobFile -Content $Params["djoin_blob"]
+    Write-FileFromBase64 -File $blobFile -Content $DJoinBlob
     $cmd = @("djoin.exe", "/requestODJ", "/loadfile", $blobFile, "/windowspath", $env:SystemRoot, "/localos")
     Invoke-JujuCommand -Command $cmd
     Invoke-JujuReboot -Now
```

Comment:
We should not trust that this function will not be misused. This has the potential to enter a reboot loop if not enclosed in a proper ```if``` statement.